src/validator/test/MutateValidatorTest.cpp

```diff
@@ -164,7 +164,7 @@ TEST_F(MutateValidatorTest, UpdateVertexTest) {
     {
         auto cmd = "UPDATE VERTEX ON person \"Tom\""
                    "SET age = age + 1 "
-                   "WHEN page == 18 "
```

Comment:
This don't report error?

Reply:
The previous implementation will not report an error.
I added the deduceType check after rewrited this Expression.